File: ls-app/src/main/java/pl/poznan/ww/ls/LsFile.java
package pl.poznan.ww.ls;

public class LsFile {
    
    private char type;
    private String canonicalPath;
    private String name;
    private String modDate;
    private Long size;
    private String attr;
    
    public LsFile(char type, String canonicalPath, String name, String modDate, Long size, String attr) {
        
        canonicalPath = canonicalPath.replaceAll("/", "&");
        canonicalPath = canonicalPath.replaceAll("\\\\", "&");
        
        this.type = type;
        this.canonicalPath = canonicalPath;
        this.name = name;
        this.modDate = modDate;
        this.size = size;
        this.attr = attr;
    }

    public char getType() {
        return type;
    }

    public void setType(char type) {
        this.type = type;
    }

    public String getCanonicalPath() {
        return canonicalPath;
    }

    public void setCanonicalPath(String canonicalPath) {
        this.canonicalPath = canonicalPath;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getModDate() {
        return modDate;
    }

    public void setModDate(String modDate) {
        this.modDate = modDate;
    }

    public Long getSize() {
        return size;
    }

    public void setSize(Long size) {
        this.size = size;
    }

    public String getAttr() {
        return attr;
    }

    public void setAttr(String attr) {
        this.attr = attr;
    }
    
}

File: ls-web/src/app/model/lsfile.ts
export class LsFile {
    private _type: string;
    private _canonicalPath: string;
    private _name: string;
    private _modDate: string;
    private _size: number;
    private _attr: string;

    get type(): string {
        return this._type;
    }
    set type(newType) {
      this._type = newType;
    }

    get canonicalPath(): string {
        return this._canonicalPath;
    }
    set canonicalPath(newCanonicalPath) {
      this._canonicalPath = newCanonicalPath;
    }

    get name(): string {
        return this._name;
    }
    set name(newName) {
      this._name = newName;
    }

    get modDate(): string {
        return this._modDate;
    }
    set modDate(newModDate) {
      this._modDate = newModDate;
    }

    get size(): number {
        return this._size;
    }
    set size(newSize) {
      this._size = newSize;
    }

    get attr(): string {
        return this._attr;
    }
    set attr(newAttr) {
      this._attr = newAttr;
    }

}
